### ISSUE TITLE
DiscoveryV1.deleteTrainingData incorrectly reports an error even if op is successful

### ISSUE THREAD

**Issue by radua-ibm** [watson-developer-cloud/node-sdk#883]
**Overview**
`DiscoveryV1.deleteTrainingData` incorrectly reports an error 
```
Error: Response not received. Body of error is HTTP ClientRequest object
```
expecting a response body while the DELETE call against `/discovery/api/v1/environments/{environment_id}/collections/{collection_id}/training_data?version=2019-03-25` is sucessful and documented to return HTTP 204 and no response body.

**Expected behavior**
`DiscoveryV1.deleteTrainingData` should not return an error when the service returns HTTP 204

**Actual behavior**
Returns an error  ^

**How to reproduce**
Refer to documentation sample : 
https://cloud.ibm.com/apidocs/discovery?code=node#delete-all-training-data

**Screenshots**

**SDK Version**
4.0.1

**Additional information:**
Node 8.9.3 or 10.11.0

**Additional context**

**Comment by dpopp07** [comment]
I recently noticed this as well - it is happening for a few Discovery methods. We will look into this very soon

**Comment by dpopp07** [comment]
This will be resolved in #888
